src/main/java/entities/Matcher.java

```diff
@@ -0,0 +1,104 @@
+package entities;
+
+import data_access_storage.UserRepo;
+import data_access_storage.UserRepoManager;
+import email_request.*;
+import matching_system.*;
+
+import javax.swing.*;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Matcher {
+//    private HashMap<User, Integer> matchResults = new HashMap<User, Integer>();
+    /** The matched result, arranged by matching score from high to low */
+    private final UserAccount[] matches;
```

Comment:
Well this variable doesn't change anyways, so it does not really matter. I made it final just because IntelliJ's warnings are annoying.

Reply:
> I'm not sure why we need to make matches as a final instance variable.

Well this variable doesn't change anyways, so it does not really matter. I made it final just because IntelliJ's warnings are annoying.

Reply:
ok